ev3dev/core.py

```diff
@@ -1810,6 +1806,51 @@ def is_pressed(self):
 
         return self.value(0)
 
+    @property
+    def is_released(self):
+        return not self.is_pressed
+
+    def _wait(self, wait_for_press, timeout_ms):
+        tic = time.time()
+
+        if self._poll is None:
+            self._value0 = self._attribute_file_open('value0')
+            self._poll = select.poll()
+            self._poll.register(self._value0, select.POLLPRI)
+
+        while True:
+            self._poll.poll(timeout_ms)
+
+            if wait_for_press:
+                if self.is_pressed:
+                    return True
+            else:
+                if self.is_released:
+                    return True
+
+            if timeout_ms is not None and time.time() >= tic + timeout_ms / 1000:
+                return False
+
+    def wait_for_pressed(self, timeout_ms=None):
+        return self._wait(True, timeout_ms)
+
+    def wait_for_released(self, timeout_ms=None):
+        return self._wait(False, timeout_ms)
+
+    def wait_for_bump(self, timeout_ms=None):
+        """
+        Wait for the touch sensor to be pressed down and then released.
+        Both actions must happen within timeout_ms.
+        """
+        start_time = time.time()
+
+        if self.wait_for_pressed(timeout_ms):
+            timeout_ms -= int((time.time() - start_time) * 1000)
```

Comment:
What happens here is timeout_ms is None? I think something like:
```py
>>> None - 4
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
```

Reply:
yep, good catch! Fixed now